File: 백준/Bronze/28062. 준석이의 사탕 사기/준석이의 사탕 사기.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
	public static void main (String[] args) throws IOException {
	   BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	   
	   String line1 = br.readLine();
	   int n = Integer.parseInt(line1), odd_cnt = 0, min_odd = 1001, sum = 0;
	   
	   StringTokenizer st = new StringTokenizer(br.readLine());
	   
	   for(int i = 0; i < n; i++){
	        int num = Integer.parseInt(st.nextToken());
	        sum += num;
	        
	        if(num % 2 == 1){
	            odd_cnt++;
	            min_odd = Math.min(min_odd, num);
	        }
	   }
	   
	   if(odd_cnt % 2 == 1){
	       sum -= min_odd;
	   }
	   
	   System.out.println(sum);
	}
}

File: 백준/Bronze/28062. 준석이의 사탕 사기/준석이의 사탕 사기.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n, num, sum = 0, odd_cnt = 0, min_odd = 1001;
    cin >> n;
    
    for(int i = 0; i < n; i++){
        cin >> num;
        sum += num;
        
        if(num % 2 == 1){
            odd_cnt += 1;
            min_odd = min(min_odd, num);
        }
    }
    
    if(odd_cnt % 2 == 1){
        sum -= min_odd;
    }
    
    cout << sum;
}